sphinx/source/start_network.rst

```diff
@@ -73,6 +73,32 @@ Other members are then allowed to vote for the proposal, using the proposal ID r
 
 The user is successfully added once a :term:`quorum` of members have accepted the proposal (``"result":true"``).
 
+Register the Lua application
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note:: This section only applies when deploying Lua applications (i.e. using the ``libluageneric.so.signed`` enclave library). For C++ applications, this step should be skipped.
+
+Before opening the CCF network to users, members should vote to register the Lua application defining the user-specific business logic (see for example :ref:`Logging (Lua)`):
+
+.. code-block:: bash
+
+    $ memberclient --cert member1_cert --privk member1_privk --rpc-address rpc_ip:rpc_port --ca network_cert update_lua_app --lua-app-file /path/to/lua/app_script
+    {"commit":9,"global_commit":8,"id":0,"jsonrpc":"2.0","result":{"completed":false,"id":1},"term":2}
+
+Other members are then allowed to vote for the proposal, using the proposal ID returned to the proposer member (here ``1``, as per ``"result":{"completed":false,"id":1}``).
+
+.. code-block::
+
+    $ memberclient --cert member2_cert --privk member2_privk --rpc-address rpc_ip:rpc_port --ca network_cert vote --proposal-id 1 --accept
+    {"commit":11,"global_commit":10,"id":0,"jsonrpc":"2.0","result":{"completed":false,"id":1},"term":2}
+
+    $ memberclient --cert member3_cert --privk member3_privk --rpc-address rpc_ip:rpc_port --ca network_cert vote --proposal-id 1 --accept
+    {"commit":13,"global_commit":12,"id":0,"jsonrpc":"2.0","result":{"completed":true,"id":1},"term":2}
+
+The Lua application is successfully registered once a :term:`quorum` of members have accepted the proposal (``"result":true"``).
+
+.. warning:: Once the network is open, it is still possible for members to update the Lua application using the same command. Note that the previous RPC handlers that are not explicitly updated in the new version of the application will still be available.
```

Comment:
I think this behaviour is odd, and since we don't have a use case so it should be removed. Rather than `update_lua_app` I think it should be `set_lua_app`, and this should first remove all existing scripts then add the newly proposed ones. We may want more finely grained `add_lua_handler` and `remove_lua_handler` in future, but for now its more sensible to reason about an all-or-nothing replacement.

Reply:
I agree with this. From a user perspective, I believe it's better to see the Lua app as a whole rather than a collection of individual endpoints.

Reply:
Yeah we definitely don't want a franken-app made of a mix of handlers.